src/sdk/namespace_fs.js

```diff
@@ -2578,12 +2578,24 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * _delete_path_dirs deletes all the paths in the hierarchy that are empty after a successful delete
+     * if the original file_path to be deleted is a directory object we want it to delete the directory
+     * if dir is not empty it will stop at the first non empty dir
+     * if dir is also a directory object (disabled mode optimization - CONTENT_DIR xattr is on the directory) we stop the loop
+     * @param {String} file_path 
+     * @param {nb.NativeFSContext} fs_context 
+     */
     async _delete_path_dirs(file_path, fs_context) {
         try {
-            let dir = path.dirname(file_path);
-            while (dir !== this.bucket_path) {
-                await nb_native().fs.rmdir(fs_context, dir);
-                dir = path.dirname(dir);
+            let dir_path = path.dirname(file_path);
+            while (dir_path !== this.bucket_path) {
+                if (file_path !== path.join(dir_path, '/') && file_path !== path.join(dir_path, config.NSFS_FOLDER_OBJECT_NAME)) {
```

Comment:
what about nested directory objects. if there are two empty content dir keys: one key dir/, and another dir/key1/.
if we delete dir/key1/ won't we delete dir/